zap/src/parser/convert.rs

```diff
@@ -1,142 +1,97 @@
 
-impl<'src> SyntaxConfig<'src> {
-	fn into_config(self, state: &mut ConvertState<'src>) -> Config<'src> {
-		let mut write_checks = false;
-		let mut typescript = false;
-
-		let mut server_output = None;
-		let mut client_output = None;
-
-		let mut casing = Casing::Pascal;
-
-		for opt in self.opts {
-			match opt.name.into_config() {
-				"write_checks" => match opt.value.kind {
-					SyntaxOptValueKind::Bool(value) => write_checks = value.into_config(),
-
-					_ => state.push_report(Report::AnalyzeInvalidOptValue {
-						span: opt.value.span(),
-						expected: "boolean",
-					}),
-				},
-
-				"typescript" => match opt.value.kind {
-					SyntaxOptValueKind::Bool(value) => typescript = value.into_config(),
-
-					_ => state.push_report(Report::AnalyzeInvalidOptValue {
-						span: opt.value.span(),
-						expected: "boolean",
-					}),
-				},
-
-				"server_output" => match opt.value.kind {
-					SyntaxOptValueKind::Str(value) => server_output = Some(value.into_config()),
-
-					_ => state.push_report(Report::AnalyzeInvalidOptValue {
-						span: opt.value.span(),
-						expected: "string",
-					}),
-				},
-
-				"client_output" => match opt.value.kind {
-					SyntaxOptValueKind::Str(value) => client_output = Some(value.into_config()),
-
-					_ => state.push_report(Report::AnalyzeInvalidOptValue {
-						span: opt.value.span(),
-						expected: "string",
-					}),
-				},
-
-				"casing" => match opt.value.kind {
-					SyntaxOptValueKind::Str(value) => match value.into_config() {
-						"PascalCase" => casing = Casing::Pascal,
-						"camelCase" => casing = Casing::Camel,
-						"snake_case" => casing = Casing::Snake,
-
-						_ => state.push_report(Report::AnalyzeInvalidOptValue {
-							span: opt.value.span(),
-							expected: "`PascalCase`, `camelCase`, or `snake_case`",
-						}),
-					},
-
-					_ => state.push_report(Report::AnalyzeInvalidOptValue {
-						span: opt.value.span(),
-						expected: "`PascalCase`, `camelCase`, or `snake_case`",
-					}),
-				},
-
-				_ => state.push_report(Report::AnalyzeUnknownOptName { span: opt.name.span() }),
-			}
+		for evdecl in config.decls.iter().filter_map(|decl| match decl {
+			SyntaxDecl::Ev(evdecl) => Some(evdecl),
+			_ => None,
+		}) {
+			evdecls.push(self.evdecl(evdecl, &tydecls));
 		}
 
-		let mut tydecls = Vec::new();
-		let mut evdecls = Vec::new();
+		if evdecls.is_empty() {
+			self.report(Report::AnalyzeEmptyEvDecls);
+		}
 
-		for decl in self.decls {
-			match decl {
-				SyntaxDecl::Ty(tydecl) => tydecls.push(tydecl.into_config(state)),
-				SyntaxDecl::Ev(evdecl) => evdecls.push(evdecl.into_config(state)),
+		let (write_checks, _) = self.boolean_opt("write_checks", true, &config.opts);
+		let (typescript, _) = self.boolean_opt("typescript", false, &config.opts);
+
+		let (server_output, _) = self.str_opt("server_output", "network/server.lua", &config.opts);
+		let (client_output, _) = self.str_opt("client_output", "network/client.lua", &config.opts);
```

Comment:
Let's make this even more explicit

```suggestion
		let (write_checks, ..) = self.boolean_opt("write_checks", true, &config.opts);
		let (typescript, ..) = self.boolean_opt("typescript", false, &config.opts);

		let (server_output, ..) = self.str_opt("server_output", "network/server.lua", &config.opts);
		let (client_output, ..) = self.str_opt("client_output", "network/client.lua", &config.opts);
```